src/Storefront/Theme/ThemeLifecycleService.php

```diff
@@ -90,15 +90,19 @@ public function __construct(
     public function refreshThemes(
         Context $context,
         ?StorefrontPluginConfigurationCollection $configurationCollection = null
-    ): void {
+    ): array {
```

Comment:
Hey @peterjaap,

this is a breaking change.
Thus, can you please change the target branch to the `major` branch?

Reply:
@PaddyS can you explain why this a breaking change is? 

As far as I can find, this method is used in three places;
- `vendor/shopware/storefront/Theme/Command/ThemeRefreshCommand.php:35`
- `vendor/shopware/storefront/Theme/Subscriber/FirstRunWizardSubscriber.php:72`
- `vendor/shopware/storefront/Theme/Subscriber/UpdateSubscriber.php:55`

None of these expect the return type to be void.

Reply:
it is a BC, because you changed the method signature. if someone has extended this class and overwrites this method, his/her extension will break

Reply:
@mitelg thanks for the explanation. I changed the target branch to major.

Reply:
Just a rebase missing and we're done here. :-)

Reply:
@PaddyS I'm getting a rebase conflict?

```
 ~/development/workspace/peterjaap/platform   patch-1  git rebase major     
First, rewinding head to replay your work on top of it...
Applying: NEXT-9278 - Dynamic product groups for Categories
Using index info to reconstruct a base tree...
M       CHANGELOG-6.3.md
M       composer.json
M       src/Administration/Resources/app/administration/test/e2e/cypress/integration/catalogue/sw-category/edit.spec.js
M       src/Core/composer.json
Falling back to patching base and 3-way merge...
Auto-merging src/Core/composer.json
Auto-merging src/Administration/Resources/app/administration/test/e2e/cypress/integration/catalogue/sw-category/edit.spec.js
Auto-merging composer.json
Auto-merging CHANGELOG-6.3.md
CONFLICT (content): Merge conflict in CHANGELOG-6.3.md
error: Failed to merge in the changes.
Patch failed at 0001 NEXT-9278 - Dynamic product groups for Categories
hint: Use 'git am --show-current-patch' to see the failed patch
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
 ✘  ~/development/workspace/peterjaap/platform  ➦ 888ec2c286 ●✚ >R>  gst             
rebase in progress; onto 888ec2c286
You are currently rebasing branch 'patch-1' on '888ec2c286'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   composer.json
        modified:   src/Administration/Resources/app/administration/src/app/component/base/sw-empty-state/index.js
        modified:   src/Administration/Resources/app/administration/src/app/component/base/sw-empty-state/sw-empty-state.html.twig
        modified:   src/Administration/Resources/app/administration/src/app/component/base/sw-empty-state/sw-empty-state.scss
        modified:   src/Administration/Resources/app/administration/src/app/component/data-grid/sw-data-grid/index.js
        modified:   src/Administration/Resources/app/administration/src/app/component/data-grid/sw-data-grid/sw-data-grid.scss
        modified:   src/Administration/Resources/app/administration/src/app/component/entity/sw-many-to-many-assignment-card/index.js
        modified:   src/Administration/Resources/app/administration/src/app/component/entity/sw-many-to-many-assignment-card/sw-many-to-many-assignment-card.html.twig
        new file:   src/Administration/Resources/app/administration/src/app/component/entity/sw-product-stream-grid-preview/index.js
        new file:   src/Administration/Resources/app/administration/src/app/component/entity/sw-product-stream-grid-preview/sw-product-stream-grid-preview.html.twig
        new file:   src/Administration/Resources/app/administration/src/app/component/entity/sw-product-stream-grid-preview/sw-product-stream-grid-preview.scss
        modified:   src/Administration/Resources/app/administration/src/app/snippet/de-DE.json
        modified:   src/Administration/Resources/app/administration/src/app/snippet/en-GB.json
        new file:   src/Administration/Resources/app/administration/src/flag/feature_next9278.js
        modified:   src/Administration/Resources/app/administration/src/module/sw-category/snippet/de-DE.json
        modified:   src/Administration/Resources/app/administration/src/module/sw-category/snippet/en-GB.json
        modified:   src/Administration/Resources/app/administration/src/module/sw-category/view/sw-category-detail-base/index.js
        modified:   src/Administration/Resources/app/administration/src/module/sw-category/view/sw-category-detail-base/sw-category-detail-base.html.twig
        modified:   src/Administration/Resources/app/administration/src/module/sw-category/view/sw-category-detail-base/sw-category-detail-base.scss
        new file:   src/Administration/Resources/app/administration/test/app/component/base/sw-empty-state.spec.js
        new file:   src/Administration/Resources/app/administration/test/app/component/data-grid/sw-data-grid.spec.js
        new file:   src/Administration/Resources/app/administration/test/app/component/entity/sw-product-stream-grid-preview.spec.js
        modified:   src/Administration/Resources/app/administration/test/core/helper/__snapshots__/sanitizer.helper.spec.js.snap
        new file:   src/Administration/Resources/app/administration/test/e2e/cypress/fixtures/product-stream-valid.json
        modified:   src/Administration/Resources/app/administration/test/e2e/cypress/integration/catalogue/sw-category/edit.spec.js
        new file:   src/Administration/Resources/app/administration/test/module/sw-category/view/sw-category-detail-base.spec.js
        modified:   src/Core/Content/Category/CategoryDefinition.php
        modified:   src/Core/Content/Category/CategoryEntity.php
        modified:   src/Core/Content/DependencyInjection/product.xml
        modified:   src/Core/Content/Product/SalesChannel/Listing/ProductListingRoute.php
        modified:   src/Core/Content/ProductStream/ProductStreamDefinition.php
        modified:   src/Core/Content/ProductStream/ProductStreamEntity.php
        modified:   src/Core/Content/Test/Product/SalesChannel/ProductListingTest.php
        new file:   src/Core/Flag/feature_next9278.php
        new file:   src/Core/Migration/Migration1592837424AddProductTypeToCategory.php
        modified:   src/Core/composer.json

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
        both modified:   CHANGELOG-6.3.md

```

Reply:
I'd suggest to checkout the major branch again and cherry-pick only your commits then. :-)